### PR TITLE
Lock the other mechanisms when climbing

### LINKED ISSUE #165: Force the robot into a specific state before climbing
We probably want to make sure that the intake is up, injector shooter and inclinator are not moving. To do this I think we want to black hole our state machines, in which case we need to consider how to recover from an accidental early trigger (perhaps hitting the shoot or intake button restarts our state machines?)

### PR DESCRIPTION
**Pull request by mlists**
closes #165

### CODE REVIEW

**Review by auscompgeek — APPROVED** [review]
lgtm, we may want to improve the shooter behaviour later

File: components/climber.py [hunk 1]
```diff
@@ -70,6 +75,15 @@ def enable_soft_limits(self) -> None:
...
         )
 
+    def is_real_climb(self) -> bool:
+        # Climbs in the last 20 seconds are real climbs...
+        return (
+            wpilib.DriverStation.getMatchTime() < 20 and self.seen_deploy_limit_switch
+        )
+
+    def should_lock_mechanisms(self) -> bool:
+        return self.is_real_climb() or not self.has_climb_finished()
```
Comment (SebZanardo) [inline components/climber.py:85]: maybe we want to change this line to `return self.is_real_climb() and self.has_climb_finished()`
and remove the self.seen_deploy_limit_switch variable

**Review by SebZanardo — APPROVED** [review]
🦾

File: components/climber.py [hunk 2]
```diff
@@ -70,6 +75,12 @@ def enable_soft_limits(self) -> None:
             CANSparkMax.SoftLimitDirection.kReverse, True
         )
 
+    def should_lock_mechanisms(self) -> bool:
+        # Climbs in the last 20 seconds are real climbs...
+        return (
+            wpilib.DriverStation.getMatchTime() < 20 and self.seen_deploy_limit_switch
+        )
```
Comment (LucienMorey) [inline components/climber.py:82]:
```suggestion
        return self.seen_deploy_limit_switch
```

File: components/climber.py [hunk 3]
```diff
@@ -97,7 +108,11 @@ def execute(self) -> None:
         if self.has_climb_finished():
             if self.last_position is not self.POSITION.RETRACTED:
                 self.status_lights.no_note()
+            if wpilib.DriverStation.getMatchTime() < 20:
```
Comment (LucienMorey) [inline components/climber.py:111]:
```suggestion
            if wpilib.DriverStation.getMatchTime() > 20:
```

**Pull request merged**